cg/services/illumina_services/illumina_metrics_service/illumina_metrics_service.py

```diff
@@ -141,8 +146,58 @@ def create_sample_sequencing_metrics_for_flow_cell(
                     sample_internal_id=sample_internal_id,
                     lane=lane,
                     metrics_parser=metrics_parser,
-                    instrument_run_id=instrument_run_id,
                     store=store,
                 )
                 sample_lane_sequencing_metrics.append(metrics)
         return sample_lane_sequencing_metrics
+
+    @staticmethod
+    def create_illumina_sequencing_run(
+        flow_cell_dir_data: FlowCellDirectoryData,
+    ) -> IlluminaSequencingRun:
+        metrics_parser = BCLConvertMetricsParser(flow_cell_dir_data.path)
+        total_reads: int = metrics_parser.get_total_reads_for_flow_cell()
+        total_undetermined_reads: int = metrics_parser.get_undetermined_reads_for_flow_cell()
+        percent_undetermined_reads: float = (
+            metrics_parser.get_percent_undetermined_reads_for_flow_cell()
+        )
+        percent_q30: float = metrics_parser.get_mean_percent_q30_for_flow_cell()
+        mean_quality_score: float = metrics_parser.get_mean_quality_score_sum_for_flow_cell()
+        total_yield: int = metrics_parser.get_yield_for_flow_cell()
+        yield_q30: int = metrics_parser.get_yield_q30_for_flow_cell()
+        cycles: int = flow_cell_dir_data.run_parameters.get_read_1_cycles()
+        software_service = IlluminaDemuxVersionService()
+        demux_software: str = software_service.get_demux_software(
+            flow_cell_dir_data.demultiplex_software_info_path
+        )
+        demux_software_version: str = software_service.get_demux_software_version(
+            flow_cell_dir_data.demultiplex_software_info_path
+        )
+        sequencing_started_at: datetime = flow_cell_dir_data.sequencing_started_at
+        sequencing_comleted_at: datetime = flow_cell_dir_data.sequencing_completed_at
+        demultiplexing_started_at: datetime = flow_cell_dir_data.demultiplexing_started_at
+        demultiplexing_completed_at: datetime = flow_cell_dir_data.demultiplexing_completed_at
+        sequencer_type: str = flow_cell_dir_data.sequencer_type
+        sequencer_name: str = flow_cell_dir_data.machine_name
+
+        return IlluminaSequencingRun(
```

Comment:
pretty sure you need to pass a device here

Reply:
It seems not since we don't try to add anything to the db... have a look at the latest implentation and tests.